source/documentation/adrs/adr-008.html.md.erb

```diff
@@ -0,0 +1,89 @@
+
+- Automatic PR environments
+- Manual PR environments
+- Other persistent environments
+
+### Production environment
+
+This is what our users access and it contains our real data. It must is the only environment in the production account.
+
+### Preproduction environment
+
+This has identical infrastructure to the production environment and is the only environment in the preproduction account. It is only accessible from within the MOJ Digital VPN. It is deployed in the pipeline before production to test integration that the release can be deployed successfully.
+
+It is be available at `preproduction.{production host}`.
+
+### Demo environment
+
+This is used to test, demo and explore functionality against the currently deployed release. It is in the development account, does not contain production data and is only accessible from within the MOJ Digital VPN. It is deployed in the pipeline after production to ensure it reflects the latest releaase.
+
+It is connected to the demo environment of any dependencies, and is the target of the demo environemnt of any dependent services. This allows cross-service testing.
+
+It is be available at `demo.{production host}`.
+
+### Local environment
+
+All services must be available to stand-up on departmental laptops in some form for local development and testing. Local environments should not require any external connections to dependent services (though their build pipeline may require external connection to e.g. fetch code and packages).
+
+### Automatic PR environments
+
+These are created automatically when a pull request is raised to use for testing and demo of changes before they are merged. They are in the development account and are only accessible from within the MOJ Digital VPN.
+
+These environments should be used to test against, rather than using local containerised environments on the CI runner, to provide a higher level of confidence.
+
+When the pull request is merged, the associated environment is automatically destroyed.
+
+They are available at `{automatically derived name}.development.{production host}`
+
+### Manual PR environments
+
+In some services it may be impractical to automatically create environments for every pull request. In this case, it should be possible to manually create an environment based on the pull request. These environments should have a "deletion date" on creation and automatically be destroyed when that date is passed.
+
+They are available at `{manually or automatically derived name}.development.{production host}`
+
+In all other regards, they are the same as automatic PR environments.
+
+## Other persistent environments
+
+Some services may require other persistent environments. For example, for training or user research. These environments may be created automatically or manually, depending on the team's needs.
+
+These environments are in the development account and do not contain production data.
+
+## Known exceptions
+
+...
```

Comment:
If we make a decision that we think is best, but we know cannot be met by any existing services, I thought we might like to document that here.

Reply:
I think we should set up or update the page for env urls for services in confluence and leave this as a decision. Any exception would be an ADR in the project's repo then